dev/archery/archery/integration/datagen.py

```diff
@@ -939,6 +962,34 @@ def generate_column(self, size, name=None):
         return StructColumn(name, size, is_valid, field_values)
 
 
+class RunLengthEncodedField(Field):
+
+    def __init__(self, name, values_field, *, nullable=True,
+                 metadata=None):
+        super().__init__(name, nullable=nullable,
+                         metadata=metadata)
+        self.run_ends_field = RunEndsField('run_ends', 32, nullable=False)
+        self.values_field = values_field
+
+    def _get_type(self):
+        return OrderedDict([
+            ('name', 'runlengthencoded')
+        ])
+
+    def _get_children(self):
+        return [
+            self.run_ends_field.get_json(),
+            self.values_field.get_json()
+        ]
+
+    def generate_column(self, size, name=None):
+        values = self.values_field.generate_column(size)
+        run_ends = self.run_ends_field.generate_column(size)
+        if name is None:
+            name = self.name
+        return RunLengthEncodedColumn(name, size, run_ends, values)
```

Comment:
shouldn't the `size` here be larger than the `size` used in `generate_column` since it would be the "logical" length for the parent?